packages/e2e/src/factories.ts

```diff
@@ -354,53 +350,52 @@ export const getWallet = async (props: GetWalletProps) => {
   const { wallet } = await setupWallet({
```

Comment:
Is there a reason to not instantiate the logger here:

``` ts
  const logger = props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger;
```
Then pass it to the `keyManagementFactory` and provider factories?

Reply:
The key managers can not be injected with a logger yet (I believe), but the factory interface requires it, so for now I am giving it a dummy logger.

Reply:
This is what I mean:
``` ts
Index: packages/e2e/src/factories.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/e2e/src/factories.ts b/packages/e2e/src/factories.ts
--- a/packages/e2e/src/factories.ts	(revision 84ca9940c0a2af72475c1686c006f9240e64ed3a)
+++ b/packages/e2e/src/factories.ts	(date 1659243657335)
@@ -347,11 +347,12 @@
  * @param props Wallet configuration parameters.
  */
 export const getWallet = async (props: GetWalletProps) => {
+  const logger = props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger;
   const { wallet } = await setupWallet({
     createKeyAgent: await keyManagementFactory.create(
       props.env.KEY_MANAGEMENT_PROVIDER,
       updateParamsId(props.idx !== undefined ? props.idx : 0, props.env.KEY_MANAGEMENT_PARAMS),
-      dummyLogger
+      logger
     ),
     createWallet: async (keyAgent: any) =>
       new SingleAddressWallet(
@@ -360,39 +361,39 @@
           assetProvider: await assetProviderFactory.create(
             props.env.ASSET_PROVIDER,
             props.env.ASSET_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           chainHistoryProvider: await chainHistoryProviderFactory.create(
             props.env.CHAIN_HISTORY_PROVIDER,
             props.env.CHAIN_HISTORY_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           keyAgent,
           networkInfoProvider: await networkInfoProviderFactory.create(
             props.env.NETWORK_INFO_PROVIDER,
             props.env.NETWORK_INFO_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           rewardsProvider: await rewardsProviderFactory.create(
             props.env.REWARDS_PROVIDER,
             props.env.REWARDS_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           stakePoolProvider: await stakePoolProviderFactory.create(
             props.env.STAKE_POOL_PROVIDER,
             props.env.STAKE_POOL_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           stores: props.stores,
           txSubmitProvider: await txSubmitProviderFactory.create(
             props.env.TX_SUBMIT_PROVIDER,
             props.env.TX_SUBMIT_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           utxoProvider: await utxoProviderFactory.create(
             props.env.UTXO_PROVIDER,
             props.env.UTXO_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           )
         }
       )
```

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/cd838c89d9b72f7032883f2c52dd0ac5cc18d551